src/engine/SpatialJoinAlgorithms.cpp

```diff
@@ -431,16 +433,21 @@ std::array<bool, 2> SpatialJoinAlgorithms::isAPoleTouched(
 
 // ____________________________________________________________________________
 Result SpatialJoinAlgorithms::BoundingBoxAlgorithm() {
-  bool alreadyWarned = false;
-
-  auto printWarning = [&]() {
-    if (!alreadyWarned) {
-      AD_LOG_WARN << "expected a point here, but no point is given. Skipping "
-                     "this point and future 'non points' of this query"
-                  << std::endl;
-      alreadyWarned = true;
-    }
-  };
+  auto printWarning =
+      [alreadyWarned = false](std::optional<SpatialJoin*> spatialJoin) mutable {
+        if (!alreadyWarned) {
+          std::string warning =
+              "The input to a spatial join contained at least one element, "
+              "that is not a point geometry and is thus skipped. Note that "
+              "QLever currently only accepts point geometries for the "
+              "spatial joins";
+          AD_LOG_WARN << warning << std::endl;
+          alreadyWarned = true;
+          if (spatialJoin) {
+            spatialJoin.value()->addWarning(warning);
```

Comment:
```suggestion
          if (spatialJoin.has_value()) {
            AD_CORRECTNESS_CHECK(spatialJoin.value() != nullptr);
            spatialJoin.value()->addWarning(warning);
```
Better to be explicit + safe.